### PR TITLE
[Icons] Add InMemory registry to allow easier tests

### PR DESCRIPTION
**Pull request by smnandre**
Will come handy to test the IconRenderer

### CODE REVIEW

**Comment by kbond** [comment]
As discussed in Slack, do we want to make this a configuration option?

Allowing this to be used as the only registry in your test environment could save a ton of http requests for apps that extensively use the _on demand_ system.

**Comment by smnandre** [comment]
Yep that was on my mind..  But i rather go step by step there, as i'm sure i need the InMemoryRegistry to add feature + test the Renderer :)

You ok if we add the config/test thing in another PR ?

File: src/Icons/tests/Unit/Registry/InMemoryIconRegistryTest.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace Symfony\UX\Icons\Tests\Unit\Registry;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\UX\Icons\Registry\InMemoryIconRegistry;
+use Symfony\UX\Icons\Svg\Icon;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ */
+final class InMemoryIconRegistryTest extends TestCase
```
Comment (norkunas) [inline src/Icons/tests/Unit/Registry/InMemoryIconRegistryTest.php:22]: `throw new IconNotFoundException(sprintf('Icon "%s" not found.', $name));`
is not covered :)

**Comment by kbond** [comment]
Could you show an example of how you intend this to be used by end users?

**Comment by kbond** [comment]
Thanks Simon.

**Comment by smnandre** [comment]
> Could you show an example of how you intend this to be used by end users?

First example there (internal so that's not the answer you expected ... but it's a start)

https://github.com/symfony/ux/pull/1606

**Pull request merged**